Library/Homebrew/os/mac.rb

```diff
@@ -12,7 +12,19 @@ module Mac
     # This can be compared to numerics, strings, or symbols
     # using the standard Ruby Comparable methods.
     def version
-      @version ||= Version.new(MACOS_VERSION)
+      @version ||= Version.new(full_version.to_s[/10\.\d+/])
+    end
+
+    # This can be compared to numerics, strings, or symbols
+    # using the standard Ruby Comparable methods.
+    def full_version
+      @full_version ||= Version.new(
+        if RUBY_PLATFORM =~ /darwin/
```

Comment:
I agree that we should keep os relate code in `os`. It just feels weird to have OS detect code in `OS::Mac`, which should means OS detection is finished. 

How about put them outside of `global.rb` into `Library/os.rb` as
```ruby 
if OS.mac?
  require "os/mac"
  ::MACOS_FULL_VERSION = OS::Mac.full_version.to_s
  ::MACOS_VERSION = OS::Mac.version.to_s
  ::OS_VERSION = "OS X #{MACOS_FULL_VERSION}"
  ISSUES_URL = ...
...